src/Microsoft.AspNetCore.SignalR.Client.Core/HubConnection.cs

```diff
@@ -128,8 +128,7 @@ public IDisposable On(string methodName, Type[] parameterTypes, Func<object[], o
                 Log.Starting(_logger);
 
                 // Start the connection
-                var connection = _connectionFactory();
-                await connection.StartAsync(_protocol.TransferFormat);
+                var connection = await _connectionFactory.ConnectAsync(_protocol.TransferFormat);
```

Comment:
Why isn't connection factory just creating the connection and StartAsync being called afterwards here? I'd like to understand more because my gut reaction is I'm not a fan of the factory creating and starting.

Reply:
Because we're deleting IConnection and ConnectionContext isn't getting StartAsync and StopAsync. IConnectionFactory (we can rename it if that makes you feel better) needs to both create and Start the ConnectionContext.

Reply:
Ok. I haven't got a better name than IConnectionFactory.

What do you think about ConnectAsync having IHubProtocol as the argument instead of just TransferFormat?

Reply:
Too much coupling? What else would it do with the IHubProtocol?

Reply:
I think this is the right contract.

Reply:
Yeah. Plus IHubProtocol is pub internal. I just worry about what happens if ConnectAsync needs more information in the future but the interface limitation is TransferFormat as the only argument.

Maybe some kind of ConnectContext class that has TransferFormat as a property. Or am I worrying too much?

Reply:
First, I understand the concern now. But lets try to think of something realistic we'd want to add here first. It would need to be something that couldn't be communicated via the construction of the factory itself.  

A context doesn't seem unreasonable.